pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1562,6 +1562,12 @@ public void connectionFailed(PulsarClientException exception) {
                     log.info("[{}] Producer creation failed for producer {} after producerTimeout", topic, producerId);
                 }
                 setState(State.Failed);
+                // release resource if the produce is going to be cleaned
+                try {
+                    this.close();
+                } catch (PulsarClientException e) {
+                    log.warn("[{}] Close a failed produce {} failed", topic, producerId, e);
```

Comment:
why do we need this ?
it is not good to add this as part of this patch, it is unrelated

Reply:
This is because the test https://github.com/apache/pulsar/blob/master/pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TopicDoesNotExistsTest.java#L68.

IF auto-create is not allowed
Before this pull request, building a producer will get a `CommandError` from the broker which will close the produce. 
With this pull request, we will get a `connectionFailed` after lookup the brokerUrl, and this failure does not close the produce which will failed the test.

I'll remove the test in this pull request, and open an new issue for the connectionFailed processing.